zebra-consensus/src/block/check.rs

```diff
@@ -245,9 +257,17 @@ pub fn miner_fees_are_valid(
     // https://zips.z.cash/protocol/protocol.pdf#txnconsensus
     let left = (transparent_value_balance - sapling_value_balance - orchard_value_balance)
         .map_err(|_| SubsidyError::SumOverflow)?;
-    let right = (block_subsidy + block_miner_fees).map_err(|_| SubsidyError::SumOverflow)?;
+    let right = (block_subsidy + block_miner_fees - expected_deferred_amount)
+        .map_err(|_| SubsidyError::SumOverflow)?;
+
+    let should_allow_unclaimed_subsidy =
+        NetworkUpgrade::current(network, height) < NetworkUpgrade::Nu6;
 
-    if left > right {
+    if if should_allow_unclaimed_subsidy {
+        left > right
+    } else {
+        left != right
```

Comment:
```suggestion
        // TODO: Document the need for claiming all subsidy once the ZIP for NU6 is out.
        left != right
```

Reply:
Applied in https://github.com/ZcashFoundation/zebra/pull/8694/commits/b8d4a7975f32eae2bc125c7da3a7be6ff5bacea9.